components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/webapp/password-recovery-notify.jsp

```diff
@@ -104,7 +104,7 @@
         var infoModel = $("#infoModel");
         infoModel.modal("show");
         infoModel.on('hidden.bs.modal', function () {
-            location.href = "<%= URLDecoder.decode(callback, "UTF-8")%>";
+            location.href = "<%= URLDecoder.decode(Encode.forJavaScriptBlock(callback), "UTF-8")%>";
```

Comment:
Change this to `Encode.forJavaScriptBlock(URLDecoder.decode(callback, "UTF-8"))`. Change other relevant locations as well.